perf/samples.js

```diff
@@ -2,9 +2,7 @@
 
   function Samples(count) {
     this.startTime = 0;
-
-    this.data = new Array(count).fill(0);
-    this.data.length = 0;
+    this.data = [];
```

Comment:
This isn't quite the same. The old code actually allocated an array up front, then set its length to zero. That would prevent allocations for during the test for re-sizing the array is it grows.